azkaban-common/src/main/java/azkaban/executor/ExecutionController.java

```diff
@@ -49,7 +49,7 @@ public ExecutionController(final Props azkProps, final ExecutorLoader executorLo
       final CommonMetrics commonMetrics,
       final ExecutorApiGateway apiGateway, final AlerterHolder alerterHolder, final
   ExecutorHealthChecker executorHealthChecker) {
-    super(azkProps, executorLoader, commonMetrics, apiGateway, alerterHolder);
+    super(azkProps, executorLoader, commonMetrics, apiGateway, alerterHolder, new DummyEventListener());
```

Comment:
Won't this break the READY or PREPARING event that you are sending from Abstract class?